xds/internal/resolver/xds_resolver_test.go

```diff
@@ -38,165 +37,79 @@ import (
-
-type s struct {
-	grpctest.Tester
-}
-
-func Test(t *testing.T) {
-	grpctest.RunSubTests(t, s{})
-}
-
-func (s) TestRegister(t *testing.T) {
-	if resolver.Get(Scheme) == nil {
-		t.Errorf("scheme %v is not registered", Scheme)
-	}
-}
-
-// testClientConn is a fake implemetation of resolver.ClientConn that pushes
-// state updates and errors returned by the resolver on to channels for
-// consumption by tests.
-type testClientConn struct {
-	resolver.ClientConn
-	stateCh *testutils.Channel
-	errorCh *testutils.Channel
-}
-
-func (t *testClientConn) UpdateState(s resolver.State) error {
-	// Tests should ideally consume all state updates, and if one happens
-	// unexpectedly, tests should catch it. Hence using `Send()` here.
-	t.stateCh.Send(s)
-	return nil
-}
-
-func (t *testClientConn) ReportError(err error) {
-	// When used with a go-control-plane management server that continuously
-	// resends resources which are NACKed by the xDS client, using a `Replace()`
-	// here simplifies tests which will have access to the most recently
-	// received error.
-	t.errorCh.Replace(err)
-}
-
-func (t *testClientConn) ParseServiceConfig(jsonSC string) *serviceconfig.ParseResult {
-	return internal.ParseServiceConfig.(func(string) *serviceconfig.ParseResult)(jsonSC)
-}
-
-func newTestClientConn() *testClientConn {
-	return &testClientConn{
-		stateCh: testutils.NewChannel(),
-		errorCh: testutils.NewChannel(),
-	}
-}
-
-// TestResolverBuilder_ClientCreationFails tests the case where xDS client
-// creation fails, and verifies that xDS resolver build fails as well.
-func (s) TestResolverBuilder_ClientCreationFails(t *testing.T) {
-	// Override xDS client creation function and return an error.
-	origNewClient := newXDSClient
-	newXDSClient = func() (xdsclient.XDSClient, func(), error) {
-		return nil, nil, errors.New("failed to create xDS client")
-	}
-	defer func() {
-		newXDSClient = origNewClient
-	}()
-
-	// Build an xDS resolver and expect it to fail.
-	builder := resolver.Get(Scheme)
+// Tests the case where xDS client creation is expected to fail because the
+// bootstrap configuration is not specified. The test verifies that xDS resolver
+// build fails as well.
+func (s) TestResolverBuilder_ClientCreationFails_NoBootstap(t *testing.T) {
+	// Build an xDS resolver without specifying bootstrap env vars.
+	builder := resolver.Get(xdsresolver.Scheme)
 	if builder == nil {
-		t.Fatalf("resolver.Get(%v) returned nil", Scheme)
+		t.Fatalf("Scheme %q is not registered", xdsresolver.Scheme)
 	}
-	if _, err := builder.Build(target, newTestClientConn(), resolver.BuildOptions{}); err == nil {
-		t.Fatalf("builder.Build(%v) succeeded when expected to fail", target)
+
+	target := resolver.Target{URL: *testutils.MustParseURL("xds:///target")}
+	if _, err := builder.Build(target, nil, resolver.BuildOptions{}); err == nil {
+		t.Fatalf("xds Resolver Build(%v) succeeded when expected to fail, because there is not bootstrap configuration for the xDS client", target)
```

Comment:
Whoops there's actualy a lot of Build(....) comments in this file. I'm fine with either one actually, I don't think we need to add it to all.

Reply:
Got to a couple more.